test/node-rpc-test.js

```diff
@@ -5,59 +5,512 @@
+      const lines = response.split('\n');
+
+      assert(response);
+      assert(lines.length);
+
+      assert.strictEqual(lines[0], 'Select a command:');
+    });
+
+    it('should contain node rpc method', async () => {
+      const response = await nclient.execute('help', []);
+      const lines = response.split('\n');
+
+      assert(response);
+      assert(lines.length);
+
+      // methods that are less likely to be depracated soon.
+      assert(lines.includes('getblockchaininfo'));
+
+      // server methods
+      assert(lines.includes('help'));
+      assert(lines.includes('stop'));
+    });
+
+    it('should not contain wallet rpc methods', async () => {
+      const response = await nclient.execute('help', []);
+      const lines = response.split('\n');
+
+      assert(response);
+      assert(lines.length);
+
+      // methods that are less likely to be depracated soon.
+      assert(!lines.includes('getwalletinfo'));
+    });
+
+    it('should get method help', async () => {
+      await assert.rejects(async () => {
+        await nclient.execute('help', ['getblockchaininfo']);
+      }, {
+        name: 'Error',
+        message: 'getblockchaininfo'
+      });
+
+      await assert.rejects(async () => {
+        await nclient.execute('help', ['help']);
+      }, {
+        name: 'Error',
+        type: 'RPCError',
+        message: 'help ( "command" )'
+      });
+    });
+
+    it('should return error on wrong command', async () => {
+      const wrongCommand = 'wrongcommand';
+
+      await assert.rejects(async () => {
+        await nclient.execute('help', [wrongCommand]);
+      }, {
+        name: 'Error',
+        message: `Method not found: ${wrongCommand}.`
+      });
+    });
 
-    await assert.rejects(async () => {
-      await nclient.execute('help', ['getinfo']);
-    }, {
-      name: 'Error',
-      message: /^getinfo/
+    it('should return help on wrong command', async () => {
+      await assert.rejects(async () => {
+        await nclient.execute('help', ['command1', 'command2']);
+      }, {
+        name: 'Error',
+        type: 'RPCError',
+        message: 'help ( "command" )'
+      });
     });
   });
 
-  it('should rpc getinfo', async () => {
-    const info = await nclient.execute('getinfo', []);
-    assert.strictEqual(info.blocks, 0);
+  describe('gettxoutproof/verifytxoutproof', function() {
+    const TEST_WALLET = 'wallet-gettxoutproof';
+
+    const blockhashes = [];
+    const txids = [];
+
+    let mempoolTXID;
+
+    before(async () => {
+      // create wallet for testing.
+      await wclient.createWallet(TEST_WALLET);
+      await wclient.execute('selectwallet', [TEST_WALLET]);
+      const testAddr = await wclient.execute('getnewaddress');
+
+      // back to miner
+      await wclient.execute('selectwallet', [MINER_WALLET]);
+      const minerAddr = await wclient.execute('getnewaddress');
+
+      { // block 0 - This block will have 2 txs (including coinbase)
```

Comment:
Technically this is block `102` (by height)...

Reply:
Not necessarily, the test is self contained, it should not matter how many blocks are/will be mined in other(future) tests. `block 0` is relative to the test as well as `blockhashes` and `txids` array.

If we add tests before gettxoutproof test, this should not start failing.